pkg/sql/catalog/accessor.go

```diff
@@ -58,18 +52,4 @@ type Accessor interface {
 	GetObjectNamesAndIDs(
 		ctx context.Context, txn *kv.Txn, db DatabaseDescriptor, scName string, flags tree.DatabaseListFlags,
 	) (tree.TableNames, descpb.IDs, error)
-
-	// GetObjectDesc looks up an object by name and returns both its
-	// descriptor and that of its parent database. If the object is not
-	// found and flags.required is true, an error is returned, otherwise
-	// a nil reference is returned.
-	//
-	// TODO(ajwerner): clarify the purpose of the transaction here. It's used in
-	// some cases for some lookups but not in others. For example, if a mutable
-	// descriptor is requested, it will be utilized however if an immutable
-	// descriptor is requested then it will only be used for its timestamp and to
-	// set the deadline.
-	GetObjectDesc(
-		ctx context.Context, txn *kv.Txn, db, schema, object string, flags tree.ObjectLookupFlags,
-	) (ResolvedObjectPrefix, Descriptor, error)
```

Comment:
Getting rid of the `Accessor` interface fell out of scope of this already tentacular PR. In all likelihood it needs to be replaced by _something_. Doing it properly is a task in and of itself, and I've come to the conclusion that it's best to do it properly. In the meantime, I've removed the methods that don't really need to be in the interface itself.